state/errors.go

```diff
@@ -52,6 +52,11 @@ type (
 		Height int64
 	}
 
+	ErrPrunerFailedToGetRetainHeight struct {
+		Which string
```

Comment:
In Rust, you'd be told to create an enum with a `Display` impl for this field. In Go, it's probably OK :stuck_out_tongue:

Reply:
Yeah in Go we'd just create a newtype (based on `string`) to represent the options and define a bunch of constants for each of the variants. It's a bit cumbersome but we could always do that in future if necessary.